longtailstorelib/remotestore.go

```diff
@@ -384,49 +382,43 @@ func remoteWorker(
 				select {
 				case _ = <-flushMessages:
 					flushReplyMessages <- 0
-				case putMsg := <-putBlockMessages:
-					if accessType == ReadOnly {
-						putMsg.asyncCompleteAPI.OnComplete(longtaillib.EACCES)
-						continue
+				case putMsg, more := <-putBlockMessages:
+					if more {
+						if accessType == ReadOnly {
+							putMsg.asyncCompleteAPI.OnComplete(longtaillib.EACCES)
+							continue
+						}
+						err := putStoredBlock(ctx, s, client, blockIndexMessages, putMsg.storedBlock)
+						putMsg.asyncCompleteAPI.OnComplete(longtaillib.ErrorToErrno(err, longtaillib.EIO))
+					} else {
+						run = false
 					}
-					err := putStoredBlock(ctx, s, client, blockIndexMessages, putMsg.storedBlock)
-					putMsg.asyncCompleteAPI.OnComplete(longtaillib.ErrorToErrno(err, longtaillib.EIO))
 				case getMsg := <-getBlockMessages:
 					fetchBlock(ctx, s, client, getMsg)
 				case prefetchMsg := <-prefetchBlockChan:
 					prefetchBlock(ctx, s, client, prefetchMsg)
-				case _ = <-stopMessages:
-					run = false
 				}
 			} else {
 				select {
 				case _ = <-flushMessages:
 					flushReplyMessages <- 0
-				case putMsg := <-putBlockMessages:
-					if accessType == ReadOnly {
-						putMsg.asyncCompleteAPI.OnComplete(longtaillib.EACCES)
-						continue
+				case putMsg, more := <-putBlockMessages:
+					if more {
+						if accessType == ReadOnly {
+							putMsg.asyncCompleteAPI.OnComplete(longtaillib.EACCES)
+							continue
+						}
+						err := putStoredBlock(ctx, s, client, blockIndexMessages, putMsg.storedBlock)
+						putMsg.asyncCompleteAPI.OnComplete(longtaillib.ErrorToErrno(err, longtaillib.EIO))
```

Comment:
Is it intended that the loop keeps going after handling this error ? (Here and in a few other places)

Reply:
Yes, that is intentional, it send the error back to the requester which handles the error.